plugins/destination/postgresql/client/write.go

```diff
@@ -3,127 +3,268 @@ package client
+	}
+	r.Status = pgtype.Status(c.Status)
+	for _, d := range c.Dimensions {
+		r.Dimensions = append(r.Dimensions, pgtype.ArrayDimension{Length: d.Length, LowerBound: d.LowerBound})
+	}
+	return r
+}
 
-	columns := make([]string, 0, len(data))
-	values := make([]interface{}, 0, len(data))
+func CQCIDRArray(c *cqtypes.CIDRArray) pgtype.CIDRArray {
+	r := pgtype.CIDRArray{}
+	for _, v := range c.Elements {
+		r.Elements = append(r.Elements, pgtype.CIDR{IPNet: v.IPNet, Status: pgtype.Status(v.Status)})
+	}
+	r.Status = pgtype.Status(c.Status)
+	for _, d := range c.Dimensions {
+		r.Dimensions = append(r.Dimensions, pgtype.ArrayDimension{Length: d.Length, LowerBound: d.LowerBound})
+	}
+	return r
+}
 
-	// Sort the columns prior to adding data to columns and values arrays
-	// Columns need to be in the same order so that the query can be cached during the statement preparation stage
-	keys := make([]string, 0, len(data))
-	for k := range data {
-		keys = append(keys, k)
+func CQMacaddrArray(c *cqtypes.MacaddrArray) pgtype.MacaddrArray {
+	r := pgtype.MacaddrArray{}
+	for _, v := range c.Elements {
+		r.Elements = append(r.Elements, pgtype.Macaddr{Addr: v.Addr, Status: pgtype.Status(v.Status)})
 	}
-	sort.Strings(keys)
-	for _, key := range keys {
-		columns = append(columns, pgx.Identifier{key}.Sanitize())
-		values = append(values, data[key])
+	r.Status = pgtype.Status(c.Status)
+	for _, d := range c.Dimensions {
+		r.Dimensions = append(r.Dimensions, pgtype.ArrayDimension{Length: d.Length, LowerBound: d.LowerBound})
 	}
-	sb.WriteString("insert into ")
-	sb.WriteString(table)
-	sb.WriteString(" (")
-	sort.Strings(columns)
-	for i, c := range columns {
-		sb.WriteString(c)
-		// sb.WriteString("::" + SchemaTypeToPg())
-		if i < len(columns)-1 {
-			sb.WriteString(",")
-		} else {
-			sb.WriteString(") values (")
-		}
+	return r
+}
+
+func CQInetArray(c *cqtypes.InetArray) pgtype.InetArray {
+	r := pgtype.InetArray{}
+	for _, v := range c.Elements {
+		r.Elements = append(r.Elements, pgtype.Inet{IPNet: v.IPNet, Status: pgtype.Status(v.Status)})
 	}
-	for i := range values {
-		sb.WriteString(fmt.Sprintf("$%d", i+1))
-		if i < len(values)-1 {
-			sb.WriteString(",")
-		} else {
-			sb.WriteString(")")
+	r.Status = pgtype.Status(c.Status)
+	for _, d := range c.Dimensions {
+		r.Dimensions = append(r.Dimensions, pgtype.ArrayDimension{Length: d.Length, LowerBound: d.LowerBound})
+	}
+	return r
+}
+
+func (c *Client) transformValues(table *schema.Table, values cqtypes.CQTypes) []interface{} {
+	pgValues := make([]interface{}, len(values))
+	for i, v := range values {
+		switch t := v.(type) {
+		case *cqtypes.Bool:
+			pgValues[i] = pgtype.Bool{
+				Bool:   t.Bool,
+				Status: pgtype.Status(t.Status),
+			}
+		case *cqtypes.Int8:
+			pgValues[i] = pgtype.Int8{
+				Int:    t.Int,
+				Status: pgtype.Status(t.Status),
+			}
+		case *cqtypes.Float8:
+			pgValues[i] = "0.0"
```

Comment:
Maybe I'm missing something, but `Float8` looks like it still needs to be implemented

Reply:
Good catch. Fixed